lib/Zonemaster/Backend/DB.pm

```diff
@@ -367,6 +364,9 @@ sub get_test_params {
 sub get_batch_job_result {
     my ( $self, $batch_id ) = @_;
 
+    die Zonemaster::Backend::Error::ResourceNotFound->new( message => "Unknown batch", data => { batch_id => $batch_id } )
+        unless defined $self->batch_exists_in_db( $batch_id );
```

Comment:
Couldn't we just look at the number of keys in %result to make this determination without and extra SQL query?

Reply:
I did this at first, then after the [discussion here](https://github.com/zonemaster/zonemaster-backend/pull/965#discussion_r835040107) I added the commit https://github.com/zonemaster/zonemaster-backend/pull/965/commits/bc96642dca8a34b2a6a24ebc6be6b7147969bc60. I can easily remove it if we think it is better to avoid an extra call to the database (which was my first thought when starting to work on this PR).

Reply:
The number of lookups based on batch ID will always be small.

Reply:
Yeah, most likely. What's at stake here is an occasional extra SQL query and about 10 extra lines of code (ignoring any documentation and unit tests). I don't think that's huge and I won't quarrel over it.

Reply:
If the check could be done more efficiently some other way it could also be fine if the magic in that solution is documented in the code.